### PR TITLE
[CM H1] fixed mount plugin tutorial, as mentioned on tuwel

### LINKED ISSUE #3722: Mount example broken
The [mount configuration files](https://github.com/ElektraInitiative/libelektra/blob/master/doc/tutorials/mount.md) example is broken.

In particular this part doesn't work:

>
> ```sh
> # mount the backend with the plugins ...
> kdb mount example.ni user:/example ni type
> 
> # ... and set a value for the demonstration
> kdb set user:/example/enumtest/fruit apple
> #> Create a new key user:/example/enumtest/fruit with string "apple"
> ```
>
> [...]
>
>```sh
> kdb meta-set user:/example/enumtest/fruit check/type enum
> kdb meta-set user:/example/enumtest/fruit check/enum "#2"
> kdb meta-set user:/example/enumtest/fruit check/enum/#0 apple
> kdb meta-set user:/example/enumtest/fruit check/enum/#1 banana
> kdb meta-set user:/example/enumtest/fruit check/enum/#2 grape
> kdb set user:/example/enumtest/fruit tomato
> # RET:5
> # this fails because tomato is not in the list of valid values
> ```

Instead of the expected results we get:

```sh
kdb meta-set user:/example/enumtest/fruit check/type enum
#> Sorry, module type issued the error C03200:
#> Validation Semantic: The type 'enum' failed to match for 'user:/example/enumtest/fruit' with string: 'apple'
#> No values are allowed (check/enum is an empty array, or parent isn't set to last element)
```

The problem here is, that with the `kdb meta-set ... check/type enum` line, we tell the `type` plugin to validate this key against an enum definition. However, we haven't set any enum values yet, so we basically tell `type` to ensure the key doesn't have a value. But we have already set a value for the key in question.

There are two solutions:

1. Don't call `kdb set` before all the `kdb meta-set` lines. That way, the key doesn't have a value when type validates against the empty enum.
2. Don't call `kdb meta-set ... check/type enum` before all the other `kdb meta-set` lines. Here the `type` plugin will only start validating the key when the enum definition is complete.

The example should be adapted to one of these two solutions.

### PR DESCRIPTION
**Pull request by philippoppel**
Tuwel issue: https://tuwel.tuwien.ac.at/mod/forum/discuss.php?d=243029#p598964
 
closes #3722 & also added sudo to the first command, which is necessary anyway

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [x] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Review by kodebach — APPROVED** [review]
LGTM

**Comment by mpranj** [comment]
Thank you so much for the contribution. Please fix the merge conflict by rebasing from master.

@kodebach please review and merge if this fully resolves  #3722.

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -214,6 +214,7 @@ _(@tucek)_
 - Added a short Visual Studio 2019 tutorial (/doc/tutorials/contributing-windows.md) _(Dominic Jäger)_
 - Added hint regarding WSL filesystem configuration (/doc/tutorials/contributing-windows.md) _(@tucek)_
 - Fixed broken link in yanlr-plugin readme _(@lawli3t)_
+- fixed #3722 _(Philipp Oppel)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:217]: Thank you for the PR! This should be a "short descriptions of your changes" which is understandable by itself, like "Fixed tutorial ...". The reference to the issue can be there additionally.

**Comment by mpranj** [comment]
Any news here?

This PR needs a rebase and has open review points.

**Comment by philippoppel** [comment]
Hey, I rebased from master and added a better description, is it fine now?

**Comment by mpranj** [comment]
Thank you! Your PR looks good, but there are still merge conflicts.

Are you sure you rebased from current upstream/master and not an outdated master branch? See the [docs](https://github.com/ElektraInitiative/libelektra/blob/master/doc/GIT.md#working-with-forks) for hints how to work with and sync forks.

**Review by mpranj — CHANGES_REQUESTED** [review]
Seems like something went wrong. There is a merge commit which should not be there when rebasing and the PR contains about 390 unrelated commits.

I created #3805, which is a rebased copy of this PR, as an example how it should look.

**Comment by philippoppel** [comment]
thank you for fixing this, sorry for the hassle

when I tried rebasing to upstream/master it did work, but when I tried to push it, my IDE wanted me to merge otherwise my push was rejected. As far as I understood it, I should have reverted the merges I did before and make like a "clean" rebase, correct?

**Comment by mpranj** [comment]
Thank you for the PR and the documentation fixes! It was merged via #3805. 

> I should have reverted the merges I did before and make like a "clean" rebase, correct?

Afair I simply copied your branch and then rebased from upstream/master again and this already removed the merge commit. Then I fixed some merge errors that were done before (duplicate news entries).
If **nothing** else works another method is: create a new branch starting from upstream/master and then `git cherry-pick` all relevant commits. This is usually the last resort though.

**Pull request closed without merging**